script/src/syscalls/write.rs

```diff
@@ -0,0 +1,60 @@
+use crate::syscalls::{INVALID_FD, SPAWN_YIELD_CYCLES_BASE, WRITE};
+use crate::types::{Fd, FdArgs, Message, VmId};
+use ckb_vm::{
+    registers::{A0, A1, A2, A7},
+    Error as VMError, Memory, Register, SupportMachine, Syscalls,
+};
+use std::sync::{Arc, Mutex};
+
+#[derive(Debug)]
+pub struct Write {
+    id: VmId,
+    message_box: Arc<Mutex<Vec<Message>>>,
+}
+
+impl Write {
+    pub fn new(id: VmId, message_box: Arc<Mutex<Vec<Message>>>) -> Self {
+        Self { id, message_box }
+    }
+}
+
+impl<Mac: SupportMachine> Syscalls<Mac> for Write {
+    fn initialize(&mut self, _machine: &mut Mac) -> Result<(), VMError> {
+        Ok(())
+    }
+
+    fn ecall(&mut self, machine: &mut Mac) -> Result<bool, VMError> {
+        if machine.registers()[A7].to_u64() != WRITE {
+            return Ok(false);
+        }
+        let fd = Fd(machine.registers()[A0].to_u64());
+        let buffer_addr = machine.registers()[A1].to_u64();
+        let length_addr = machine.registers()[A2].to_u64();
+        let length = machine
+            .memory_mut()
+            .load64(&Mac::REG::from_u64(length_addr))?
+            .to_u64();
+
+        // We can only do basic checks here, when the message is actually processed,
+        // more complete checks will be performed.
+        // We will also leave to the actual write operation to test memory permissions.
+        if !fd.is_write() {
+            machine.set_register(A0, Mac::REG::from_u8(INVALID_FD));
+            return Ok(true);
```

Comment:
Should it return `Ok` here? Or `Err`?

Reply:
It should return Ok(true), which indicates that the syscall has been executed normally. This is the common way of writing syscalls, such as https://github.com/nervosnetwork/ckb/blob/31bcecddc8ad5b7d0a30399ef55a5ce0a359bb7e/script/src/syscalls/load_header.rs# L165-L168